stdlib/public/stubs/CMakeLists.txt

```diff
@@ -18,6 +18,18 @@ set(LLVM_OPTIONAL_SOURCES
     ${swift_stubs_objc_sources}
     ${swift_stubs_unicode_normalization_sources})
 
+# ICU isn't required on Darwin, but is on every other platform.
+# Now in case we're cross-compiling from Darwin for another platform,
+# the find_package should still be executed.
+if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+  set(icu_required, "REQUIRED")
+else()
+  set(icu_required, "")
+endif()
+
+find_package(ICU ${icu_required} COMPONENTS uc)
+set(ICU_UC_LIBRARY "")
```

Comment:
Having looked at the Android and Windows solution, it's difficult for me to find the proper way forward.

- If I understand @hughbe correctly, the Windows side doesn't rely on find_package() to gather include and library flags.
- If I understand @modocache 's code in PR #5432 correctly, it doesn't seem like the Android side relies on the include flags for ICU.

In other words, both examples pointed to don't seem to provide solutions for the problem on FreeBSD (or other platforms for which the compiler doesn't have a default include path set where ICU can be found). find_package() provides the correct include paths on FreeBSD, but the include flags aren't used for Android and Windows.

It would be far easiest to take the Android solution route and simply add -I/usr/local/include somewhere (e.g. in AddSwift.cmake), but that doesn't really feel like a good solution given that find_package() would provide the correct path and not some hardcoded path...

So, if someone could comment (@jrose-apple ?) on the route that is deemed most correct, that would be greatly appreciated.

Thanks